registry/storage/linkedblobstore.go

```diff
@@ -231,7 +231,47 @@ func (lbs *linkedBlobStore) Delete(ctx context.Context, dgst digest.Digest) erro
 	return nil
 }
 
-func (lbs *linkedBlobStore) Enumerate(ctx context.Context, ingestor func(digest.Digest) error) error {
+// LayersEnumerate find layer's digest in the Repository.
+func LayersEnumerate(ctx context.Context, storageDriver driver.StorageDriver, repoName string, ingester func(digest.Digest, string) error) error {
```

Comment:
If you add linkDirectoryPathSpec to linkedBlobStore in [repository.Blobs](https://github.com/docker/distribution/blob/1e2f10eb65743fed02f573d31a4587de09afb20e/registry/storage/registry.go#L302), you will be able to use linkedBlobStore.Enumerate instead.